BLT-881: initial cut at developer skills documentation

mikemadison13 opened the pull request:
Fixes #881 .

Changes proposed:
- updating Onboarding doc to link to Skills Doc
- Created Skills Doc that outlines minimum and suggested Skills for developers working with BLT

mikemadison13 commented:
@grasmash and @danepowell I would love eyes on this to ensure I haven't left out any resources that are worthwhile, misstated any objectives, etc.

danepowell commented:
I like the idea of centralizing this kind of documentation, however I'm wondering if there's a less pointed way of linking to it from the onboarding doc. While it's true that developers _should_ eventually gain all of these skills, they aren't absolutely necessary if all you want to do is install a local Drupal site. I just don't want to make BLT seem unapproachable for newbies.

mikemadison13 commented:
@danepowell I agree that I don't want to scare people off, but I am trying to paint a realistic picture of "can you do these things?" if yes, then great. if no, then you should brush up or you'll have trouble.

grasmash commented on readme/skills.md:
```diff
@@ -0,0 +1,126 @@
+## Enterprise Web Development
+### Virtual Environment
+#### [DrupalVM](https://www.drupalvm.com/)
+This project aims to make spinning up a simple local Drupal test/development environment incredibly quick and easy, and to introduce new developers to the wonderful world of Drupal development on local virtual machines (instead of crufty old MAMP/WAMP-based development).
```
We don't require that people use VM with BLT. In fact, I use MAMP :)

I'd remove this or note that it isn't required.

grasmash commented on readme/skills.md:
```diff
@@ -0,0 +1,126 @@
+ * [composer require](https://getcomposer.org/doc/03-cli.md#require)
+ 
+####[NPM](https://www.npmjs.com/)
+npm is the package manager for JavaScript. Find, share, and reuse packages of code from hundreds of thousands of developers — and assemble them in powerful new ways.
```
This is not part of BLT. It is an aspect of Cog. We need to indicate that this isn't required. Perhaps a separate "optional" section.

grasmash commented on readme/skills.md:
```diff
@@ -0,0 +1,126 @@
+ * [Netbeans](http://netbeans.org/features/php/)
+
+##Front End Technologies
+###[Gulp](http://gulpjs.com/)
```
Likewise this is Cog specific.

grasmash commented on readme/skills.md:
```diff
@@ -0,0 +1,126 @@
+  * [git rebase -i](https://git-scm.com/docs/git-rebase#git-rebase---interactive)
+
+### Github
+GitHub is a web-based Git repository hosting service. It offers all of the distributed version control and source code management (SCM) functionality of Git as well as adding its own features. It provides access control and several collaboration features such as bug tracking, feature requests, task management, and wikis for every project.
```
GitHub isn't required by BLT.

mikemadison13 commented:
@grasmash I have updated the language about a virtual environment (adding dev desktop, mentioning that it and drupal vm are optional). I also moved all of the NPM stuff / github down into the additional reading section (updated it to be optional) and added a note for all the front end stuff to reference cog and say it's not part of blt but is often bundled with it and its commands.
